BSP/STM32F413/Src/BSP_Fans.c

```diff
@@ -81,52 +18,24 @@ Inputs: Number of fan to change speed (1-4)
         Speed of Fan(0-8)
 */
 ErrorStatus BSP_Fans_Set(uint8_t fan, uint32_t speed){
+    //don't mess with the contactor
+    if (fan == CFAN) return ERROR; 
     //Range of pulse is 0-4000
     //First check to make sure that change is within range of values
     //Load new value into Compare and Capture Register
-    if  (speed < 0 || speed > TOPSPEED) return ERROR; ; //if not in range, leave function with error code
-    switch (fan)
-    {
-    case 1:
-        TIM_SetCompare1(TIM8, (speed * DIVIDER));
-        return SUCCESS;
-        break;
-    case 2:
-        TIM_SetCompare2(TIM8, (speed * DIVIDER));
-        return SUCCESS;
-        break;
-    case 3:
-        TIM_SetCompare1(TIM12, (speed * DIVIDER));
-        return SUCCESS;
-        break;
-    case 4:
-        TIM_SetCompare2(TIM12, (speed * DIVIDER));
-        return SUCCESS;
-        break;
-    default:
-        return ERROR; //invalid fan value
-        break;
-    }
+    if  (speed < 0 || speed > TOPSPEED) return ERROR; //if not in range, leave function with error code
+    
+    return BSP_PWM_Set(fan, speed * DIVIDER);
 }
 
 int BSP_Fans_GetSpeed(uint8_t fan){
-    switch (fan)
-    {
-    case 1:
-        return TIM_GetCapture1(TIM8) / DIVIDER;
-        break;
-    case 2:
-        return TIM_GetCapture2(TIM8) / DIVIDER;
-        break;
-    case 3:
-        return TIM_GetCapture1(TIM12) / DIVIDER;
-        break;
-    case 4:
-        return TIM_GetCapture2(TIM12) / DIVIDER;
-        break;
+    if(fan == CFAN) return -1;
```

Comment:
```suggestion
    if (fan == CFAN) {
        return -1;
    }
    else {
        return BSP_PWM_Get(fan);
    }
```

Reply:
Idk how to format correctly using GitHub comments, but the above would probably be a better way of writing this function.

Reply:
Or u could turn the whole thing into a nested ternary operator, its just nicer to have if else instead of if if.